### DISCUSSION TITLE
Can't record my screen with Screen Studio when AeroSpace is enabled

### DISCUSSION THREAD

**Discussion by emiliosheinz** [nikitabobko/AeroSpace#993] [potential-bugs] [answered]
First of all, thank you for building such a great tool. I'm really enjoying it. 

My setup has 3 monitor (the MacBook's built-in and two external Dell ones). When AeroSpace is enabled and I try to start a recording on a specific screen with Screen Studio, the overlays that Screen Studio should be showing individually on each screen become all cluttered in a single one which makes it impossible to start the recording. 

![Screenshot 2024-08-24 at 19 52 47](https://github.com/user-attachments/assets/66ba99e4-0697-4811-8848-225ac91b4a26)

Disabling AeroSpace solves the issue but it is not ideal since I want to record the screen with it on.

Additionally, I've tried the following on my config but it doesn't work. 

```toml
[[on-window-detected]]
if.app-id = 'com.timpler.screenstudio'
check-further-callbacks = true
run = 'layout floating'
```

Based on that, I believe this could be related to (1) How AeroSpace manages screens and workspaces or (2) Aerospace interfering in the way the overlays are being rendered.

**AeroSpace version**

```
❯ aerospace --version
aerospace CLI client version: 0.14.2-Beta 0cb8dbdfc5ee73b8cbc200f175f467ebead55201
AeroSpace.app server version: 0.14.2-Beta 0cb8dbdfc5ee73b8cbc200f175f467ebead55201
```

Let me know if I van provide any additional information.

**Edit 1:**   

Further investigating it, it looks like what is happening is that AeroSpace is moving all the overlays to the focused screen which is making them cover each other.

**Comment by andrebrito16** [comment]
Having the same problem. Any solution to this?

**Reply by tebuevd** [reply]
Also interested in a solution.

**Reply by emiliosheinz** [reply]
I stopped using AeroSpace for that reason. Unfortunately, I didn't find any solution for it.

**Reply by AlienOctopus** [reply]
also experiencing this

**Reply by azamuddin** [reply]
Experiencing it right now.

**Reply by wassimk** [reply]
I had the same issue. I love AeroSpace and Screen Studio and wanted them to work well together. Quitting AeroSpace before every recording and restarting it afterward was getting old fast. 

I was looking for a better solution when I read in the docs you can _disable_ AeroSpace without losing all your window positions. Perfect! Since I already use a keyboard shortcut via [Hammerspoon](https://www.hammerspoon.org) to move and resize windows for screencasting, I just added an `aerospace enable off` command before the window resizing happens. Works like a charm. 

If I get ambitious, I could save AeroSpace's state and toggle it back on with the same shortcut, but having to re-enable it with the menu is perfectly fine. Here is my Hammerspoon [script](https://github.com/wassimk/dotfiles/blob/5bf6bf2fe100ef44c9d712aea632db20330422d9/hammerspoon/init.lua#L16-L21) for reference.

**Reply by andrebrito16** [reply]
@wassimk thanks for this! But after the creator of AeroSpace just ignore this one, I moved back to Yabai :)

**Comment by nikitabobko** [comment, marked as answer]
Created an issue: https://github.com/nikitabobko/AeroSpace/issues/1271

I don't care about this bug, PRs are welcome
